caffe2/python/onnx/backend.py

```diff
@@ -334,18 +334,40 @@ def _make_rnn_direction(cls, input_blob, B, W, R, initial_states_and_names, sequ
                             ends  =[weight_offset + 1 * gates_hidden_size,-1])
 
         initial_states_sliced = []
+        batch_size = None
+        default_init_state = None
+        for initial_state, _ in initial_states_and_names:
+            if not initial_state:
+                input_shape = pred_mh.net.Shape(input_blob, name + '/input_shape')
+                batch_size = pred_mh.net.Slice(input_shape, name + '/batch_size_slice', starts=[1], ends=[2])
+                default_initial_state = cls.dummy_name()
+                state_before_tile = pred_mh.net.ConstantFill([],
+                        [default_initial_state + '/before_tile'],
+                        shape=[1, 1, hidden_size],
+                        value=0., dtype=input_elem_type)
+                batch_axis = cls.dummy_name()
+                pred_mh.net.ConstantFill([], [batch_axis], shape=[1],
+                        value=1, dtype=core.DataType.INT32)
+                pred_mh.net.Tile([state_before_tile, batch_size, batch_axis],
+                        [default_initial_state])
+                break
+
         for initial_state, name_suffix in initial_states_and_names:
-            initial_states_sliced.append(
-                init_net.Slice(initial_state, name + name_suffix,
-                               starts=[direction_offset + 0, 0, 0],
-                               ends  =[direction_offset + 1,-1,-1]))
+            if not initial_state:
+                initial_states_sliced.append(default_initial_state)
+            else:
+                initial_states_sliced.append(
+                    init_net.Slice(initial_state, name + name_suffix,
```

Comment:
this actually needs to be pred_mh.net, see the bugfix in https://github.com/pytorch/pytorch/pull/7820

Reply:
Sure, I will change it since it's a conflict.